gamestonk_terminal/stocks/sector_industry_analysis/financedatabase_view.py

```diff
@@ -0,0 +1,280 @@
+    """
+    for industry in financedatabase_model.get_industries():
+        print(industry)
+
+
+def display_bars_financials(
+    finance_metric: str,
+    country: str,
+    sector: str,
+    industry: str,
+    marketcap: str = "",
+    exclude_exchanges: bool = True,
+    limit: int = 10,
+    export: str = "",
+):
+    """
+    Display financials bars comparing sectors, industry, analysis, countries, market cap and excluding exchanges.
+
+    Parameters
+    ----------
+    finance_metric: str
+        Select between operatingCashflow, revenueGrowth, ebitda, grossProfits, freeCashflow, earningsGrowth,
+        returnOnAssets, debtToEquity, returnOnEquity, totalCash, totalDebt, totalRevenue, quickRatio,
+        recommendationMean, ebitdaMargins, profitMargins, grossMargins, operatingMargins, totalCashPerShare
+    country: str
+        Search by country to find stocks matching the criteria.
+    sector : str
+        Search by sector to find stocks matching the criteria.
+    industry : str
+        Search by industry to find stocks matching the criteria.
+    marketcap : str
+        Select stocks based on the market cap.
+    exclude_exchanges: bool
+        When you wish to include different exchanges use this boolean.
+    limit: int
+        Limit amount of companies displayed
+    export: str
+        Format to export data as
+    """
+    stocks_data = financedatabase_model.get_stocks_data(
+        country, sector, industry, marketcap, exclude_exchanges
+    )
+
+    metric_data = {}
+    for symbol in list(stocks_data.keys()):
+        if (
+            "financialData" in stocks_data[symbol]
+            and "quoteType" in stocks_data[symbol]
+        ):
+            metric = stocks_data[symbol]["financialData"][finance_metric]
+            stock_name = stocks_data[symbol]["quoteType"]["longName"]
+            if metric:
+                metric_data[stock_name] = (metric, symbol)
+
+    if len(metric_data) > 1:
+
+        plt.subplots(figsize=plot_autoscale(), dpi=PLOT_DPI)
+        if gtff.USE_ION:
+            plt.ion()
+
+        metric_data = dict(
+            OrderedDict(
+                sorted(metric_data.items(), key=lambda t: t[1][0], reverse=True)
+            )
+        )
+
+        company_name = list()
+        company_metric = list()
+        company_ticker = list()
+        company_metric_to_do_median = list()
+        for idx, metric in enumerate(metric_data.items()):
+            if idx < limit:
+                company_name.append(metric[0])
+                company_metric.append(metric[1][0])
+                company_ticker.append(metric[1][1])
+
+            else:
+                company_metric_to_do_median.append(metric[1][0])
+
+        company_metric_to_do_median += company_metric
+
+        if company_metric_to_do_median:
+            print(f"Limiting the amount of companies displayed to {limit}.")
+
+        for n, m, t in zip(
+            company_name[::-1], company_metric[::-1], company_ticker[::-1]
```

Comment:
Perhaps this is personal preference but I dislike reading code that has a bunch of 0's and 1's in it like you are programming here. It makes it difficult to understand what is going on without debugging myself and thus reduces readability. It helps if these 1's and 0's could be changed to actual names of what you are selecting (and thus use dictionaries instead of lists for example).

Reply:
In this case the [::-1] just reverts the order of the list to be the opposite, I had to do this due to the implementation of plt.barh and me wanting to have the bigger elements on top.

This is a standard reverting operation on a list, hence why I didn't think it was worth creating a new variable just for this as it is only used here.